src/vivarium/framework/population/population_view.py

```diff
@@ -281,16 +384,120 @@ def _coerce_to_dataframe(
+        if not set(population_update.columns).issubset(view_columns):
             raise PopulationError(
                 f"Cannot update with a DataFrame or Series that contains columns "
                 f"the view does not. Dataframe contains the following extra columns: "
-                f"{set(population_update.columns).difference(self.columns)}."
+                f"{set(population_update.columns).difference(view_columns)}."
+            )
+
+        update_columns = list(population_update)
+        if not update_columns:
+            raise PopulationError(
+                "The update method of population view is being called "
+                "on a DataFrame with no columns."
             )
 
         return population_update
 
-    def __repr__(self):
-        return (
-            f"PopulationView(_id={self._id}, _columns={self.columns}, _query={self._query})"
-        )
+    @staticmethod
+    def _ensure_coherent_initialization(
+        population_update: pd.DataFrame, state_table: pd.DataFrame
+    ) -> None:
+        """Ensure that overlapping population updates have the same information.
+
+        During population initialization, each state table column should be updated by
+        exactly one component and each component with an initializer should create at
+        least one column. Sometimes components are a little sloppy and provide
+        duplicate column information, which we should continue to allow. We want to ensure
+        that a column is only getting one set of unique values though.
+
+        Parameters
+        ----------
+        population_update
+            The update to the simulation state table.
+        state_table
+            The existing simulation state table. When this method is called, the table
+            should be in a partially complete state. That is the provided population
+            update should carry some new attributes we need to assign.
+
+        Raises
+        -----
+        PopulationError
+            If the population update contains no new information or if it contains
+            information in conflict with the existing state table.
+
+        """
+        missing_pops = len(state_table.index.difference(population_update.index))
+        if missing_pops:
+            raise PopulationError(
+                f"Components should initialize the same population at the simulation start. "
+                f"A component is missing updates for {missing_pops} simulants."
+            )
+        new_columns = set(population_update).difference(state_table)
+        overlapping_columns = set(population_update).intersection(state_table)
+        if not new_columns:
+            raise PopulationError(
+                f"A component is providing a population update for {list(population_update)} "
+                "but all provided columns are initialized by other components."
+            )
+        for column in overlapping_columns:
```

Comment:
Why  do we want to allow this?

ETA: Won't we throw an error in this case later anyway in line 334?

Reply:
I was actually being too restrictive in 334 and have updated the logic to be consistent with what's here.